src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,77 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * The Deadline task which represents a deadline.
+ *
+ * @author Leong Jia Hao Daniel
+ */
+public class Deadline extends Task {
+
+    protected LocalDate date;
+    protected String time;
+    protected String dateTime;
+
+    /**
+     * The constructor for a Deadline task.
+     *
+     * @param description The description of the task.
+     * @param dateTime The data and time of the task in the d/MM/yyyy format.
+     */
+    public Deadline(String description, String dateTime) {
+        super(description);
+        String[] details = dateTime.split(" ");
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("d/MM/yyyy");
+        this.date = LocalDate.parse(details[0], inputFormat);
```

Comment:
Perhaps you might want to consider removing the `this` since the parameter you take in your constructor for Deadline has a different name.